### PR TITLE
Deprecate obsolete reactive API

### PR DESCRIPTION
**Pull request by qwwdfsad**

### CODE REVIEW

**Comment by qwwdfsad** [comment]
I've read the reactive guide again and I have a strong feeling we don't need it.
It's outdated, it operates with obsolete concepts, e.g. compares channels with publishers, and advocates for creating your own operators on top of the coroutines (one operator == one coroutine).
I'd rather remove it at all

File: README.md [hunk 1]
```diff
@@ -61,8 +61,8 @@ suspend fun main() = coroutineScope {
   * [Kotlin Coroutines in Practice](https://www.youtube.com/watch?v=a3agLJQ6vt8) (Roman Elizarov at KotlinConf 2018, [slides](https://www.slideshare.net/elizarov/kotlin-coroutines-in-practice-kotlinconf-2018))
 * Guides and manuals: 
   * [Guide to kotlinx.coroutines by example](https://kotlinlang.org/docs/reference/coroutines/coroutines-guide.html) (**read it first**)
+  * [Guide to programming with Flow](docs/flow.md)
```
Comment (elizarov) [inline README.md]: It is a part of general guide. IMO, no need to list separately.

File: docs/flow.md [hunk 2]
```diff
@@ -1794,6 +1795,18 @@ as cancellation and structured concurrency serve this purpose.
 Note that [launchIn] also returns a [Job], which can be used to [cancel][Job.cancel] the corresponding flow collection
 coroutine only without cancelling the whole scope or to [join][Job.join] it.
  
+### Flow and Reactive Streams
+
+For those who are familiar with [Reactive Streams](https://www.reactive-streams.org/) or reactive frameworks such as RxJava and project Reactor, 
+design of the Flow may look very familiar.
+
+Indeed, its design was inspired by Reactive Streams and its various implementations. But Flow main goal always was (and is) to have as simple design as possible,
```
Comment (elizarov) [inline docs/flow.md]: ` always was (and is)` -> `is`

File: docs/flow.md [hunk 3]
```diff
@@ -1794,6 +1795,18 @@ as cancellation and structured concurrency serve this purpose.
 Note that [launchIn] also returns a [Job], which can be used to [cancel][Job.cancel] the corresponding flow collection
 coroutine only without cancelling the whole scope or to [join][Job.join] it.
  
+### Flow and Reactive Streams
+
+For those who are familiar with [Reactive Streams](https://www.reactive-streams.org/) or reactive frameworks such as RxJava and project Reactor, 
+design of the Flow may look very familiar.
+
+Indeed, its design was inspired by Reactive Streams and its various implementations. But Flow main goal always was (and is) to have as simple design as possible, 
+be Kotlin and suspension friendly and respect structured concurrency. And achieving this goal would be impossible without reactive pioneers and their tremendous work. You can read the complete story in [Reactive Streams and Kotlin Flows](https://medium.com/@elizarov/reactive-streams-and-kotlin-flows-bfd12772cda4) article.
```
Comment (elizarov) [inline docs/flow.md]: `And achieving` -> `Achieving`

**Comment by elizarov** [comment]
Good riddance!

**Comment by qwwdfsad** [comment]
Merged manually

**Pull request closed without merging**